### PR TITLE
Update jest and ts-jest to latest versions

### PR DESCRIPTION
**Pull request by evantahler**
## Change description

Description here

### Development

- [ ] Application changes have been tested appropriately

### Impact

- [ ] Code follows company security practices and guidelines
- [ ] Security impact of change has been considered
- [ ] Performance impact of change has been considered
- [ ] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "prepare": "./bin/build",
     "dev": "./bin/dev",
     "pretest": "npm run lint && npm run prepare",
-    "test": "jest",
+    "test": "jest --logHeapUsage --no-cache",
```
Comment (teallarson) [inline core/package.json:29]: This will be nice to keep an eye on usage moving forward 👍

File: core/jest.config.js [hunk 2]
```diff
@@ -5,14 +5,18 @@ module.exports = {
...
     "<rootDir>/__tests__/utils",
     "<rootDir>/__tests__/fixtures",
     "<rootDir>/src/.*/*.ts",
     "<rootDir>/dist/.*/*.ts",
     "<rootDir>/dist/.*/*.js",
   ],
+  globals: {
+    "ts-jest": {
+      transpileOnly: true,
+      isolatedModules: true,
```
Comment (teallarson) [inline core/jest.config.js:19]: Wondering if we want to add this to the compiler or to other suites? Was reading some in the last comment in this thread: 
https://github.com/kulshekhar/ts-jest/issues/1967#issuecomment-1017510990

**Review by teallarson — APPROVED** [review]
I left a question on whether we want to add `isolatedModules` elsewhere as well, but other that looks good!

**Comment by evantahler** [comment]
I'm going to close this PR - the addtional logging in the test is nice, but we know that we can't fix it as of yet.  Dependabot will update the versions of jests/ts-jest in other PRs

**Pull request closed without merging**